cardano-node-chairman/test/Test/Cardano/Node/Chairman/Shelley.hs

```diff
@@ -0,0 +1,413 @@
+import           Data.Maybe
+import           Data.Ord
+import           Data.Semigroup
+import           Data.String (String)
+import           GHC.Float
+import           GHC.Num
+import           Hedgehog (Property, discover, (===))
+import           Lens.Micro
+import           Lens.Micro.Aeson
+import           System.Exit (ExitCode (..))
+import           System.IO (IO)
+import           Text.Read
+import           Text.Show
+
+import qualified Chairman.Base as H
+import qualified Chairman.IO.File as IO
+import qualified Chairman.IO.Network.Socket as IO
+import qualified Chairman.IO.Network.Sprocket as IO
+import qualified Chairman.Process as H
+import qualified Chairman.String as S
+import qualified Data.HashMap.Lazy as HM
+import qualified Data.List as L
+import qualified Data.Map.Lazy as M
+import qualified Data.Time.Clock as DTC
+import qualified Hedgehog as H
+import qualified System.Directory as IO
+import qualified System.Environment as IO
+import qualified System.FilePath.Posix as FP
+import qualified System.IO as IO
+import qualified System.Process as IO
+
+{- HLINT ignore "Redundant <&>" -}
+
+rewriteConfiguration :: String -> String
+rewriteConfiguration "Protocol: RealPBFT" = "Protocol: TPraos"
+rewriteConfiguration "minSeverity: Info" = "minSeverity: Debug"
+rewriteConfiguration "TraceBlockchainTime: False" = "TraceBlockchainTime: True"
+rewriteConfiguration s = s
+
+rewriteGenesisSpec :: Int -> Value -> Either String Value
+rewriteGenesisSpec supply v = Right $ v
+  & key "slotLength" .~ toJSON @Double 0.2
+  & key "activeSlotsCoeff" .~ toJSON @Double 0.1
+  & key "securityParam" .~ toJSON @Int 10
+  & key "epochLength" .~ toJSON @Int 1500
+  & key "maxLovelaceSupply" .~ toJSON supply
+  & key "protocolParams" . key "decentralisationParam" .~ toJSON @Double 0.7
+
+prop_spawnShelleyCluster :: Property
+prop_spawnShelleyCluster = H.propertyOnce . H.workspace "chairman" $ \tempAbsPath -> do
+  tempBaseAbsPath <- H.noteShow $ FP.takeDirectory tempAbsPath
+  tempRelPath <- H.noteShow $ FP.makeRelative tempBaseAbsPath tempAbsPath
+  base <- H.noteShowM H.getProjectBase
+  baseConfig <- H.noteShow $ base <> "/configuration/chairman/defaults/simpleview"
+  currentTime <- H.noteShowIO DTC.getCurrentTime
+  startTime <- H.noteShow $ DTC.addUTCTime 10 currentTime -- 10 seconds into the future
+  socketDir <- H.noteShow $ tempRelPath <> "/socket"
+  env <- H.evalIO IO.getEnvironment
+
+  let bftNodes = ["node-bft1", "node-bft2"] :: [String]
+  let bftNodesN = ["1", "2"] :: [String]
+  let poolNodes = ["node-pool1"] :: [String]
+  let allNodes = bftNodes <> poolNodes :: [String]
+  let numBftNodes = 3 :: Int
+
+  let userAddrs = ["user1"]
+  let poolAddrs = ["pool-owner1"]
+  let addrs = userAddrs <> poolAddrs
+
+  H.writeFile (tempAbsPath <> "/configuration.yaml") . L.unlines . fmap rewriteConfiguration . L.lines =<<
+    H.evalIO (IO.readFile (base <> "/configuration/defaults/byron-mainnet/configuration.yaml"))
```

Comment:
Similar question here, if this is a chairman for a shelley cluster, why are we reading a byron config? Why not just read a shelley config?

Reply:
The question didn't occur to me, but now that I think about it, does it make sense to use the `mainnet` config and make adjustments to it because it is known to be maintained?

Reply:
I would have used what is in `configuration/chairman/defaults/simpleview`. AFAIU there is no longer a need to test a mixed cluster of nodes (i.e we only need to test Shelley only). I would confirm this with Duncan. I think rather than modify configs on the fly it would make more sense to just keep what's in `configuration/chairman/defaults/simpleview` up to date.

Reply:
@dcoutts can you clarify if we want shelly-only or byron-shelley chairman tests or both?

I plan to soon make a similar PR based on the `byron-shelley/mkfiles.sh` script, so it would be good to know if that is actually needed.

Reply:
For a test that runs in CI on every PR we of course want to keep things short. I think that means we do not have time to run multiple epochs and cross epoch boundaries. Perhaps at most we can cross one epoch boundary (if we configure things for very short epochs).

So I think that tells us that for a short-running CI test we should use a single era consensus mode, and have that be the latest era. At the moment that is Shelley of course.

But it would also be nice to have a chairman configuration for a longer running nightly test to be able to run with multiple epochs and multiple eras, starting from Byron and crossing all our era boundaries (currently just two of course, but a third is fast approaching).

Reply:
Switching to the `simpleview` config is non-trivial, since the straightforward replacement doesn't work and it's difficult to pin down the problem with a simple `diff`.

I'd like to merge this bit as is and address the configuration file concerns in subsequent PRs, perhaps after cleaning up the config files to make diffing them easier.

Reply:
I will work on the configuration file tidy up in this PR https://github.com/input-output-hk/cardano-node/pull/1792

Reply:
In order to move away from using the `byron-mainnet` configuration file, I duplicated and customised the copy for the shelly-only chairman test, which lets me avoid a rewrite.